core/engine.go

```diff
@@ -346,6 +346,19 @@ func (e *Engine) processSamples(samples ...stats.Sample) {
 			}
 			sm.Metric.Sink.Add(sample)
 		}
+
+		if e.Options.RunTags == nil {
+			continue
+		}
+
+		if samples[i].Tags == nil {
+			samples[i].Tags = e.Options.RunTags
+			continue
+		}
+
+		for k, v := range e.Options.RunTags {
+			samples[i].Tags[k] = v
```

Comment:
Hm, I'm not sure what the precedence for the tags should be. If the user specifies a tag with `--run-tags mytag=myval1` and then has `export let options = { runTags : {mytag: "myval2"} }` and `http.get("https://httpbin.org/", { tags: { mytag: "myval3" } });` in the code, what tag value should be emitted? I think the order should be `mytag3`, if not present `mytag1`, finally (with the least priority) `mytag2`, but I'm not completely sure that's the correct approach.
@robingustafsson: thoughts?

Reply:
Hm, yeah I think the cascading order should be Request level -> CLI -> Env vars -> Options -> Config file (where -> means it overrides whatever is on the right). So more or less the same as we do already, besides the request-level case: https://github.com/loadimpact/k6/blob/master/cmd/run.go#L140

The only reason I can think of why the request level should not override all other possible option inputs is that you won't be able to override it from outside of the script (using CLI or Env vars). However, in reality that might not be an issue as one can argue that if you have specified a tag on the request level it's probably for a very good reason.

Reply:
Yeah, agree, and you can "override" tags you've specified at the request level from the CLI if you just use environment variables in the code tags :slightly_smiling_face: 

So here the for loop should check if `Tags[k]` is present before saving the value, I think that should be enough.